### DISCUSSION TITLE
Düzenli ifadeler

### DISCUSSION THREAD

**Discussion by B3Y0N4** [19/classroom#3] [Ruby] [answered]
![Screenshot 2022-01-11 213056](https://user-images.githubusercontent.com/72824981/149328237-d61488f5-5f7b-4fd5-8577-c1637b2dbe33.jpg)
ikinci ve dördüncü satir nasil çalişiyor

**Comment by roktas** [comment, marked as answer]
2'nci satırda içteki parantezlere dikkat et `/(...)/` , bu diyor ki:  "eşlenen match'i yakala (buna capture group diyoruz) bir yerde dursun (parantezlere alıyoruz), ben buna daha sonra `\1` (buna da back reference diyoruz) diyerek erişeceğim."  Sonrasında "değiştir" kısmında `(\1)` ile eşlenen metnin etrafına parantez konuluyor.  İki parantezi karıştırma, bul kısmındaki (gsub'ın ilk argümanI) parantezler gruplama için, değiştir kısmındaki (gsub'ın ikinci argümanı) parantez ise bildiğin düz metin, yerine başka bir şey de koyabilirsin (deneyerek bak).

4'ncü satır: aynı şey, tek farkı numaralı back reference'lar yerine eşlenen metne isimlerle erişebilmen sağlanıyor.  Nasıl? Bul kısmında `(...)` yerine `(?<İSİM>...)`, değiştir kısmında `\1` yerine `\k<İSİM>` diyebiliyorsun.  Bu kullanıma "named captures" deniliyor.

Daha öte bilgiler için gerekli anahtar kelimeleri verdim, onları aratırsın.
